src/frequenz/sdk/timeseries/logical_meter/_formula_steps.py

```diff
@@ -57,10 +58,9 @@ def apply(self, eval_stack: List[Optional[float]]) -> None:
         """
         val2 = eval_stack.pop()
         val1 = eval_stack.pop()
-        if val1 is None or val2 is None:
+        res = val1 + val2
+        if (isnan(res) or isinf(res)) and (len(eval_stack) == 0):
             res = None
```

Comment:
I think we don't have to do this check at the end of each step.  It can be moved to the top-level apply (in the `FormulaEngine` class, from which the `apply` of the individual steps are called),  where we can do it once, after all the steps have finished.

Reply:
Right, I was not sure about this but it makes sense as FormulaEngine knows if all steps are done. I'll update it and call for a review after that